generic-sensor/resources/generic-sensor-helpers.js

```diff
@@ -62,7 +62,8 @@ function sensor_test(func, name, properties) {
 
 function verifySensorReading(pattern, values, timestamp, isNull) {
   function round(val) {
-    return Number.parseFloat(val).toPrecision(6);
+    const res = Number.parseFloat(val).toPrecision(6);
```

Comment:
Can you add a comment explaining why we do this (something like _"If |val| cannot be converted to a float, we return the original value. This can happen when a value in |pattern| is not a number."_)?

Reply:
Done.